Undefined name 'loss' in the creation of 'training_loss_dict'

### Description

Undefined name __loss__ caused by an invalid comprehension in the code:
```python
training_loss_dict = average_sharded_losses([{
              "training": l
} for l in loss for loss in sharded_losses.values()])
```
https://github.com/tensorflow/tensor2tensor/blob/master/tensor2tensor/utils/t2t_model.py#L341-L343

The above code is an invalid comprehension and will fail with a __NameError__ just like
```
python -c '[{"training": l} for l in loss for loss in "this is a test".split()]'
```

### Environment information

```
OS: https://github.com/tensorflow/tensor2tensor

$ python -V
# 3.7.0
```

### For bugs: reproduction and error logs

# Steps to reproduce:
$ __flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics__

# Error logs:
[flake8](http://flake8.pycqa.org) testing of https://github.com/tensorflow/tensor2tensor

$ __flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics__
```
./tensor2tensor/utils/t2t_model.py:227:22: F821 undefined name 'loss'
          } for l in loss for loss in sharded_losses.values()])
                     ^
1    F821 undefined name 'loss'
1
```

--- a/tensor2tensor/utils/t2t_model.py
+++ b/tensor2tensor/utils/t2t_model.py
@@ -338,9 +338,9 @@ class T2TModel(base.Layer):
             sharded_logits[k] = dp(self.top, v, datashard_to_features)
             sharded_losses[k] = dp(self.loss, sharded_logits[k],
                                    datashard_to_features)
-          training_loss_dict = average_sharded_losses([{
+          training_loss_dict = average_sharded_losses([({
               "training": l
-          } for l in loss for loss in sharded_losses.values()])
+          } for l in loss) for loss in sharded_losses.values()])
           losses.update(training_loss_dict)
         else:
           sharded_logits = dp(self.top, body_out, datashard_to_features)

